internal/controller/controllers/agent_controller.go

```diff
@@ -132,6 +132,18 @@ func (r *AgentReconciler) updateIfNeeded(ctx context.Context, agent *adiiov1alph
 		return ctrl.Result{}, errors.New("Host not found in cluster")
 	}
 
+	internalHost, err := r.Installer.GetHostInternal(ctx, string(*c.ID), agent.Name)
+	if err != nil {
+		return ctrl.Result{Requeue: true, RequeueAfter: defaultRequeueAfterOnError}, err
+	}
+
+	if internalHost.Approved != spec.Approved {
+		err = r.Installer.UpdateHostApprovedInternal(ctx, string(*c.ID), agent.Name, spec.Approved)
```

Comment:
in case of an error, `UpdateHostApprovedInternal` may either return`http.StatusNotFound` or any other error.

In InstallEnv controller we requeue only for the former:


https://github.com/openshift/assisted-service/blob/6b56b6707c4d5b43e024a56a5827a231632ec531/internal/controller/controllers/installenv_controller.go#L109-L116

Reply:
Done